numpy/core/records.py

```diff
@@ -610,9 +610,7 @@ def fromarrays(arrayList, dtype=None, shape=None, formats=None,
         # and determine the formats.
         formats = []
         for obj in arrayList:
-            if not isinstance(obj, ndarray):
-                raise ValueError("item in the array list must be an ndarray.")
-            formats.append(obj.dtype.str)
+            formats.append(obj.dtype)
```

Comment:
Thinking again, I think this might fix a bug. Can you add a test that calls `fromarrays([some_int_array, some_structured_array])` and see if it behaves as you expect before and after this patch?

Reply:
@eric-wieser 

```python
import numpy as np

x = np.arange(10)

y = np.ones(10, dtype=[('a', '<u2'), ('b', '<i4')])

y['a'] = np.arange(10, 20, dtype='<u2')
y['b'] = np.arange(-10, -20, -1, dtype='<i4')

z = np.core.records.fromarrays([x, y])

print(z)
```

with the patch the snippet works (as I expected)

```
[(0, (10, -10)) (1, (11, -11)) (2, (12, -12)) (3, (13, -13))
 (4, (14, -14)) (5, (15, -15)) (6, (16, -16)) (7, (17, -17))
 (8, (18, -18)) (9, (19, -19))]
```

without the patch I get an exception

```
Traceback (most recent call last):

  File "<ipython-input-7-a0bf5895bdf5>", line 1, in <module>
    runfile('/home/daniel/Documents/python/profiling.py', wdir='/home/daniel/Documents/python')

  File "/home/daniel/Documents/python/venvs/v_asammdf/lib/python3.6/site-packages/spyder_kernels/customize/spydercustomize.py", line 709, in runfile
    execfile(filename, namespace)

  File "/home/daniel/Documents/python/venvs/v_asammdf/lib/python3.6/site-packages/spyder_kernels/customize/spydercustomize.py", line 108, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "/home/daniel/Documents/python/profiling.py", line 10, in <module>
    z = np.core.records.fromarrays([x, y])

  File "/home/daniel/Documents/python/venvs/v_asammdf/lib/python3.6/site-packages/numpy/core/records.py", line 644, in fromarrays
    _array[_names[i]] = arrayList[i]

ValueError: Can't cast from structure to non-structure, except if the structure only has a single field.
```

Reply:
Great news! Could you commit a regression test for that?

Reply:
Is the last commit what you had in mind?